readers/mongodb/messages.go

```diff
@@ -44,9 +47,14 @@ func (repo mongoRepository) ReadAll(chanID string, rpm readers.PageMetadata) (re
 	sortMap := map[string]interface{}{
 		order: -1,
 	}
+
+	limit := rpm.Limit
+	if rpm.Limit == noLimit {
+		limit = 0
+	}
 	// Remove format filter and format the rest properly.
 	filter := fmtCondition(chanID, rpm)
-	cursor, err := col.Find(context.Background(), filter, options.Find().SetSort(sortMap).SetLimit(int64(rpm.Limit)).SetSkip(int64(rpm.Offset)))
+	cursor, err := col.Find(context.Background(), filter, options.Find().SetSort(sortMap).SetLimit(limit).SetSkip(int64(rpm.Offset)))
```

Comment:
We don't want offset if there is no limit (at least for now). Probably better to simply have `cursor, err := col.Find(context.Background(), filter, options.Find().SetSort(sortMap)` if limit == noLimit